### PR TITLE
CAMEL-18648: Google-bigquery - fixed Exception during endpoint creation

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://issues.apache.org/jira/browse/CAMEL-18648

### CODE REVIEW

**Review by davsclaus — APPROVED** [review]

**Review by oscerd — APPROVED** [review]
Does it needs to go in 3.18.x?

**Comment by JiriOndrusek** [comment]
@oscerd 3.18.x is unaffected (refactor did not  happen there - [code](https://github.com/apache/camel/blob/camel-3.18.x/components/camel-google/camel-google-bigquery/src/main/java/org/apache/camel/component/google/bigquery/GoogleBigQueryConfiguration.java#L63))

File: components/camel-google/camel-google-bigquery/src/main/java/org/apache/camel/component/google/bigquery/GoogleBigQueryConfiguration.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public void parseRemaining(String remaining) {
         projectId = parts[0];
         datasetId = parts[1];
         if (parts.length > 2) {
-            tableId = parts[3];
+            tableId = parts[2];
```
Comment (orpiske) [inline components/camel-google/camel-google-bigquery/src/main/java/org/apache/camel/component/google/bigquery/GoogleBigQueryConfiguration.java:62]: Oops. My bad on this one. I was trying to see if the tests would catch this but I forgot to fix it. Great catch, thanks!!!

**Review by orpiske — APPROVED** [review]

File: components/camel-google/camel-google-bigquery/src/main/java/org/apache/camel/component/google/bigquery/GoogleBigQueryConfiguration.java (hunk 1 above)
Comment (JiriOndrusek) [reply components/camel-google/camel-google-bigquery/src/main/java/org/apache/camel/component/google/bigquery/GoogleBigQueryConfiguration.java:62]: Np. I upgraded test to catch it.

**Pull request merged**